### PR TITLE
feat: patches SMCP instead of creating one

### PR DESCRIPTION
**Pull request by bartoszmajsak**
- uses patches for existing SMCP instead of creating one
- enables customization of SMCP name 
- fixes authorino sidecar injection
- simplifies init-job (logic and roles)
- docs: updates requirements for SMCP installation

### CODE REVIEW

**Review by cam-garrison — APPROVED** [review]
A few minor nitpicks in my review but overall LGTM reading it over.

Testing notes:
Tested on a completely fresh CRC cluster.

- First created an SMCP named `basic` in `istio-system`. 
- Then, created KfDef in opendatahub.
- init-job ran succesfully, restarted `authorino` pod.
- However, all pods (including dashboard) in `opendatahub` were 1/1.
I believe this was because the `dashboard` pod was created before the SMM:
- dashboard `creationTimestamp: '2023-07-03T19:14:25Z'`
- SMM `creationTimestamp: "2023-07-03T19:14:54Z"`
However, I was still prompted to login and successfully reached the dashboard without the sidecar injected :confused: 
Restarted `dashboard` pod, created notebooks and was routed to them successfully with no further issues.

I'm not sure what the solution is for this, restarting dashboard pod to be safe seems like overkill.

File: enabling-ossm.md [hunk 1]
```diff
@@ -55,12 +65,48 @@ operator-sdk run bundle quay.io/cgarriso/opendatahub-operator-bundle:dev-0.0.2 -
...
+
+For example, the following commands configure a minimal profile:
+
+```sh
+kubectl create ns istio-system
+kubectl apply -n istio-system -f -<<EOF
+apiVersion: maistra.io/v2
+kind: ServiceMeshControlPlane
+metadata:
+  name: minimal
```
Comment (bartoszmajsak) [inline enabling-ossm.md:86]:
```suggestion
  name: basic
```

File: enabling-ossm.md [hunk 2]
```diff
@@ -55,12 +65,48 @@ operator-sdk run bundle quay.io/cgarriso/opendatahub-operator-bundle:dev-0.0.2 -
...
+>
+> You may need to manually finalize the installation of the Authorino operator via the Installed Operators tab in the OpenShift Console.
+
+
+> **Warning**
+>
+> Make sure to configure the Service Mesh Control Plane, as we are patching it.
+
+
+For example, the following commands configure a minimal profile:
```
Comment (bartoszmajsak) [inline enabling-ossm.md:78]:
```suggestion
For example, the following commands configure a slimmed-down profile:
```

File: enabling-ossm.md [hunk 3]
```diff
@@ -55,12 +65,48 @@ operator-sdk run bundle quay.io/cgarriso/opendatahub-operator-bundle:dev-0.0.2 -
...
+
+```sh
+kubectl create ns istio-system
+kubectl apply -n istio-system -f -<<EOF
+apiVersion: maistra.io/v2
+kind: ServiceMeshControlPlane
+metadata:
+  name: minimal
+spec:
+  version: v2.3
```
[the remark this answers is not in the document]
Comment (bartoszmajsak) [reply enabling-ossm.md]: Done in [ee2aef4](https://github.com/maistra/odh-manifests/pull/26/commits/ee2aef4a36e0dd39ea985cd9f7e446870216a9ce)

File: enabling-ossm.md [hunk 4]
```diff
@@ -150,29 +201,60 @@ kubectl create ns auth-provider
...
+
+If the latter is the case (i.e., an error like `E0328 18:39:56.277217 1 access.go:177] osin: error=unauthorized_client, internal_error=<nil> get_client=client check failed, client_id=odh`)`, check if the token is the same everywhere by comparing the output of the following commands:
+
+
+```sh
+kubectl get oauthclient.oauth.openshift.io odh
+kubectl exec $(kubectl get pods -n istio-system -l app=istio-ingressgateway  -o jsonpath='{.items[*].metadata.name}') -n istio-system -c istio-proxy -- cat /etc/istio/odh-oauth2/token-secret.yaml
+kubectl get secret istio-odh-oauth2 -n istio-system -o yaml
+```
+To read the actual value of secrets you could use a [`kubectl` plugin](https://github.com/elsesiy/kubectl-view-secret) instead. Then the last line would look as follows `kubectl view-secret istio-odh-oauth2 -n istio-system`.
```
[the remark this answers is not in the document]
Comment (bartoszmajsak) [reply enabling-ossm.md]: Thanks, done in [8f4a372](https://github.com/maistra/odh-manifests/pull/26/commits/8f4a372a3903e3374196de8c4e0c86f6723ed3ce)

File: service-mesh/authorino/base/operator-cluster-wide-no-tls.yaml [hunk 5]
```diff
@@ -3,6 +3,8 @@ kind: Authorino
 metadata:
   name: authorino
   namespace: auth-provider
+  labels:
+    sidecar.istio.io/inject: "true"
```
[the remark this answers is not in the document]
Comment (bartoszmajsak) [reply service-mesh/authorino/base/operator-cluster-wide-no-tls.yaml]: Good catch, fixed in [c5fa008](https://github.com/maistra/odh-manifests/pull/26/commits/c5fa0086e29e4503d4e51a18612af5a9a5ea47cc)

**Comment by bartoszmajsak** [comment]
Thanks for the review @cam-garrison. I am looking at the dashboard issue right now, where we lack proxy sidecar.

**Pull request merged**